buoy.js

```diff
@@ -1,42 +1,66 @@
 /**
- * Buoy initializer.
+ * Buoy main "app" file.
+ *
+ * This file contains the Buoy front-end JavaScript "root" code; all
+ * other Buoy JavaScripts are dependent on this file, so this file
+ * must be present and loaded before any other dependent files.
+ *
+ * In this file, we create a global `BUOY` object using the (classic)
+ * {@link http://www.adequatelygood.com/JavaScript-Module-Pattern-In-Depth.html|JavaScript Module Pattern}
+ * and return a simple interface to interact with {@link http://BackboneJS.org|Backbone}
+ * library components (`Models`, `Views`, and `Collections`), alon with
+ * some helper functions.
  *
  * @license GPL-3.0
  */
 
-// Determine where "in Buoy" we are, register appropriate handlers.
-jQuery(document).ready(function () {
-    BUOY.init();
-});
-
-// Respond to the "Install iOS" event (triggered by a simulated install button).
-jQuery(document).on('install.ios', function () {
-    BUOY.installWebApp();
-});
-
 /**
- * The main Buoy "module" initializes the page behavior.
+ * The main Buoy "module" is our single global variable for all Buoy
+ * front-end behavior and values. It contains all our front-end code.
  */
 var BUOY = (function () {
 
     /**
+     * Buoy's data model (using `Backbone.Model`).
+     *
+     * @var {object}
+     */
+    Models = {};
+
+    /**
+     * Buoy's view code (using `Backbone.View`).
+     *
+     * @var {object}
+     */
+    Views = {};
+
+    /**
+     * Buoy's model collections (using `Backbone.Collection`).
+     *
+     * @var {object}
+     */
+    Collections = {};
+
+    /**
      * Initializer.
      */
     var init = function () {
         // Note: This works around GitHub issue #47.
         // Could be removed after WebKit and/or Bootstrap fixes this in their libs.
-        if (jQuery('.dashboard_page_buoy_chat, .dashboard_page_buoy_activate_alert').length) {
+        if (jQuery(buoy_dom_hooks.page_chat + ' ' + buoy_dom_hooks.page_activate_alert).length) {
```

Comment:
I don't feel like we're gaining much here in terms of readability with `buoy_dom_hooks`. I would prefer a named boolean helper function like:
```JavaScript
function pageHasActiveAlert() {
  return jQuery(buoy_dom_hooks.page_chat + ' ' + buoy_dom_hooks.page_activate_alert).length > 0;
};
```

``` JavaScript
if (pageHasActiveAlert) {
  // Do the things
}
```
Such helpers could even be defined on `buoy_dom_hooks` potentially. I'm not sure if there's a more Backboney way to handle this stuff.